app/grandchallenge/pages/views.py

```diff
@@ -153,7 +153,7 @@ def delete(self, request, *args, **kwargs):
 
 def get_average_job_duration_for_phase(phase):
     algorithm_images = Submission.objects.filter(
-        phase__slug=phase.slug
+        phase__pk=phase.pk
```

Comment:
Even simpler, you can just do:

```suggestion
        phase=phase
```